File: x/week6-python/sentimental-mario-less/mario.py
from cs50 import get_int


def main():
    # Prompt the user for the Height of the pyramid
    while True:
        height = get_int("Height: ")
        if 1 <= height <= 8:
            break

    # Print the pyramid
    for i in range(1, height + 1):
        print_row(i, height)


def print_row(bricks, height):
    spaces = height - bricks

    # Print spaces
    print(spaces * " ", end="")

    # Print bricks
    print(bricks * "#")



main()

File: x/week1-c/mario/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int bricks, int height);

int main(void)
{
    // Prompt the user for the hieght
    int height;
    do
    {
        height = get_int("Height? ");
    }
    while (height < 1 || height > 8);

    // Print the pyramid??
    for (int i = 0; i < height; i++)
    {
        print_row(i + 1, height);
    }
}

void print_row(int bricks, int height)
{
    // Print the right pyramid
    int spaces = height - bricks;
    for (int i = 0; i < spaces; i++)
    {
        printf(" ");
    }
    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }

    // The space between the two pyramids
    printf("  ");

    // Print the left pyramid
    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }

    printf("\n");
}
